assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -10,40 +10,28 @@
 from azureml.fsspec import AzureMachineLearningFileSystem
 from datetime import datetime
 from dateutil import parser
+from pyspark.sql.functions import lit
 from shared_utilities.io_utils import (
     init_spark,
     read_mltable_in_spark,
     save_spark_df_as_mltable,
 )
 
+from shared_utilities.constants import MDC_CORRELATION_ID_COLUMN, MDC_DATA_COLUMN
 
-def mdc_preprocessor(
-    data_window_start: str,
-    data_window_end: str,
-    input_data: str,
-    preprocessed_input_data: str,
-):
-    """Extract data based on window size provided and preprocess it into MLTable.
 
-    Args:
-        production_data: The data asset on which the date filter is applied.
-        monitor_current_time: The current time of the window (inclusive).
-        window_size_in_days: Number of days from current time to start time window (exclusive).
-        preprocessed_data: The mltable path pointing to location where the outputted mltable will be written to.
-    """
-    format_data = "%Y-%m-%d %H:%M:%S"
-    start_datetime = parser.parse(data_window_start)
-    start_datetime = datetime.strptime(
-        str(start_datetime.strftime(format_data)), format_data
+def _format_date_string(format: str, date_to_format: str) -> datetime:
+    parsed_date = parser.parse(date_to_format)
+    return datetime.strptime(
+        str(parsed_date.strftime(format)), format
     )
 
-    # TODO Validate window size
-    end_datetime = parser.parse(data_window_end)
-    end_datetime = datetime.strptime(
-        str(end_datetime.strftime(format_data)), format_data
-    )
 
-    # Create mltable, create column with partitionFormat
+def _create_mltable_definition_to_transform_mdc_data(
```

Comment:
wondering if a name like `convert_uri_folder_to_mltable` would be more fitting here. I wasn't sure what this was doing from the name and had to read through the code to figure it out!

Reply:
Good idea - will follow-up in a bit. Thanks